### PR TITLE
API Client tests

### PR DESCRIPTION
**Pull request by HaicoDorenbos**

### CODE REVIEW

File: pkg/api/client_test.go [hunk 1]
```diff
@@ -1,12 +1,11 @@
-package api_test
+package api //_test
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: `api_test`

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rvolosatovs) [inline .idea/vcs.xml]: Remove

File: .idea/inspectionProfiles/Project_Default.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
Comment (rvolosatovs) [inline .idea/inspectionProfiles/Project_Default.xml]: Remove

File: pkg/api/client_test.go [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
 
 import (
 	"bytes"
 	"encoding/json"
 	"strconv"
 	"testing"
 
-	. "github.com/rvolosatovs/turtlitto/pkg/api"
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: Put it back, given the comment above

File: pkg/api/client_test.go [hunk 5]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
 		})
 	}
 }
+
+//----------------------------------------------
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: Useless comment

File: pkg/api/client_test.go [hunk 6]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
...
+					if err != nil {
+						panic(err)
+					}
+					return len(b), nil
+				},
+			}
+
+			cl := NewClient(in, out)
+
+			_, err := cl.SetState(tc.Expected)
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: You should check the returned state here too.

File: pkg/api/client_test.go [hunk 7]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
...
+			a.Nil(err)
+
+			s, err := cl.State()
+			a.Nil(err)
+
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: remove newline

File: pkg/api/client_test.go [hunk 8]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
...
+				WriteFunc: func(b []byte) (int, error) {
+					var m Message
+					err := json.Unmarshal(b, &m)
+					a.Nil(err)
+					a.NotEmpty(m.MessageID)
+					a.Equal(MessageTypeSetState, m.Type)
+					var ts map[string]*State
+					err = json.Unmarshal(m.Payload, &ts)
+					a.Nil(err)
+					a.Equal(ts, tc.Expected)
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: It should be `tc.Input`, which should be the same, as the value passed to `cl.SetState`, but different from `tc.Expected`, otherwise a function, which simply always returns the passed state would pass this test.

File: pkg/api/client_test.go [hunk 9]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
...
+
+					err = json.NewEncoder(out).Encode(&Message{
+						MessageID: m.MessageID,
+						Type:      m.Type,
+						Payload:   pld,
+					})
+					if err != nil {
+						panic(err)
+					}
+					return len(b), nil
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: Might want to set some boolean variable here and check it after `SetState` returns to assert that this function was actually executed.

File: pkg/api/client_test.go [hunk 10]
```diff
@@ -71,3 +71,153 @@ func TestState(t *testing.T) {
...
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			a := assert.New(t)
+
+			var writerExec bool //used for checking whether the writer is actually executed
+
+			out := &bytes.Buffer{}
+			in := &mockWriter{
+				WriteFunc: func(b []byte) (int, error) {
+					writerExec = true
```
Comment (HaicoDorenbos) [inline pkg/api/client_test.go]: Shouldn't this be placed at the end of the function?

Comment (rvolosatovs) [reply pkg/api/client_test.go]: No

File: pkg/api/client_test.go [hunk 11]
```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
...
+			s, err := cl.State()
+			a.Nil(err)
+
+			a.Equal(s, tc.Expected)
+		})
+	}
+}
+
+func TestCommand(t *testing.T) {
+	testCases := []Command{
```
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply pkg/api/client_test.go]: @Desahri ? Please stick to the same format

File: pkg/api/client_test.go [hunk 12]
```diff
@@ -71,3 +71,153 @@ func TestState(t *testing.T) {
...
+						Type:      MessageTypeCommand,
+					})
+					if err != nil {
+						panic(err)
+					}
+					return len(b), nil
+				},
+			}
+
+			err := NewClient(in, out).SendCommand(command)
```
Comment (rvolosatovs) [inline pkg/api/client_test.go]: Test execution in the same way as in the other test case.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -142,3 +142,16 @@ In case one of the words specifies the meaning of the variable in a specific lan
 Code should be as self-explanatory as possible. However, comments should be used to respect Go formatting guidelines and to explain what can not be expressed by pure code. Comments should be English sentences, and documentation-generating comments should be closed by a period. Comments can also be used to indicate steps to take in the future (_TODOs_).
 
 * In **Go files**, comments should be added according to `golint` requirements and [Effective Go guidelines](https://golang.org/doc/effective_go.html#commentary), especially in regards to commenting exported packages, types and variables.
+
+* In **Javascript files**, comments should be added when needed. Single line comments should use `//`, but when they are placed inside a styled-component, `/* */` should be used. Multi line comments should be of the `/* */` form. React components should get a multi line comment of the `/** */` form containing its purpose, the author of the component and an ordered list of props it accepts. [PropTypes validation](https://reactjs.org/docs/typechecking-with-proptypes.html) should also be added. The following can be used as a template:
```
Comment (rvolosatovs) [inline CONTRIBUTING.md]: Unrelated

Comment (Desahri) [reply CONTRIBUTING.md]: I think this went wrong when merging with master, not sure how to fix this
